### PR TITLE
Add note to hist docstring about speed

### PR DESCRIPTION
**Pull request by EJEP**
## PR Summary

Add note to hist docstring stating that using `histtype='bar'` can be slow when using a large number of bins. I have used the wording proposed here: https://github.com/matplotlib/matplotlib/pull/12493/commits. As #12493 involves adding a new rcparam and has had no comments for almost a year I have made this a separate pull request in order to resolve the discussions in #7121.
 
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -6540,6 +6540,11 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         --------
         hist2d : 2D histograms
 
+        Notes
+        -----
+        - For large numbers of bins, 'step' and 'stepfilled' can be
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: should the strings be in double backticks to render as code?

[the remark this answers is not in the document]
Comment (jklymak) [reply lib/matplotlib/axes/_axes.py]: If we are going to note this should we also note that bars may disappear if less than one pixel wide?

Comment (EJEP) [reply lib/matplotlib/axes/_axes.py]: I will leave this as is then.

[the remark this answers is not in the document]
Comment (EJEP) [reply lib/matplotlib/axes/_axes.py]: I have added (>10,000) to the note, as using the basis from #12493 this seems like a 'noticeable' difference.

[the remark this answers is not in the document]
Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: Then let's rather go for 1000 bins as a limit for two reasons:

1. For pedagogical reasons, I don't want to have values that are much larger than usual output device resolutions. Higher numbers are niche applications and I don't want to convey the idea that 10,000 bins are actually something to consider as an average user. 1000 bins are just about what one can still resolve on a usual screen.

2. `%timeit plt.hist(x, histtype='bar', bins=1000)` bins need 600ms here. This is human-perceptible as a slight delay. Also re-rendering when panning or resizing the plot starts to get a noticable delay.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -6540,6 +6540,11 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         --------
         hist2d : 2D histograms
 
+        Notes
+        -----
+        - For large numbers of bins (>1000), 'step' and 'stepfilled' can be
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]: Please remove the bullet point. There's no need to have a list with just one element here. A plain paragraph is perfectly fine.

**Comment by timhoffm** [comment]
Thanks!

**Pull request merged**